common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,24 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  def cleanAmpSoundcloud(document: Document) = {
+    document.getElementsByClass("element-audio").filterNot { audioElement: Element =>
+      audioElement.getElementsByTag("iframe").isEmpty
+    }.foreach { audioElementWithIframe: Element =>
+      audioElementWithIframe.getElementsByTag("iframe").foreach { iframeElement: Element =>
+        val soundcloudUrl = iframeElement.attr("src")
+        val pattern = ".*%2Ftracks%2F(\\d+).*".r
+        soundcloudUrl match {
+          case pattern(trackId) =>
+            val soundcloud = createAmpSoundcloudElement(document, trackId)
+            audioElementWithIframe.appendChild(soundcloud)
+            iframeElement.remove()
+          case _ =>
```

Comment:
don't necessarily listen too much to my specific suggestions, but in general, I fund a useful principle for structuring code is think about how you would explain what you want the code to do to someone else, and then make sure the code as written matches that structure.
I don't want to pick apart too much such a small piece of code, but think about  considerations such as correctness, clarity and anything else you can think of.  Some may be competing, but not always.